tools/devops/automation/templates/build/build.yml

```diff
@@ -52,7 +52,13 @@ steps:
 
 - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
   - pwsh: |
-      git reset --hard HEAD^2 
+      git config remote.origin.fetch '+refs/pull/*:refs/remotes/origin/pull/*'
+      git fetch origin
+      $branch="$(Build.SourceBranch)".Replace("merge", "head")
+      $branch=$branch.Replace("refs", "origin")
```

Comment:
Go from refs/PRID/merge to origin/PRID/head. That way we get the PR code without a merge with main.